routes/users.js

```diff
@@ -1,28 +1,24 @@
 const express = require('express')
 const router = express.Router()
 const passport = require('passport');
-
-const User = require('../models/user')
+require('../config/passport')(passport);
 
 const usersController = require('../controllers/users')
 
+function isLoggedIn(req, res, next) {
+  if (req.isAuthenticated()) return next()
+  res.redirect('/');
+}
+
 router.get('/', usersController.getIndex)
 router.get('/login', usersController.getLogin)
 router.get('/logout', usersController.getLogout)
 router.get('/signup', usersController.getSignUpForm)
-router.get('/profile', usersController.getProfile)
-router.post('/signup', function(req, res, next) {
-  const { username, password } = req.body
-
-  const newUser = new User({
-    username,
-    password
-  })
-
-  newUser.save(function(err) {
-    if (err) return next(err)
-    res.render('newpoll')
-  })
-});
+router.post('/signup', usersController.createNewUser)
+router.post('/signin', passport.authenticate('local', {
+  successRedirect: '/users/profile',
+  failureRedirect: '/users/login'
+}));
+router.get('/profile', isLoggedIn, usersController.getProfile)
```

Comment:
If you want to protect a certain endpoint, use the `isLoggedIn` function before the controller. If the user is signed in, it will proceed with the route. If not, it will redirect them to the login page.